pkg/apis/catalog/basic_view.go

```diff
@@ -40,6 +41,10 @@ func (r *CreateRequest) Validate() error {
 		return fmt.Errorf("unsupported catalog type %q", r.Type)
 	}
 
+	if _, err = regexp.Compile(r.FilterPattern); err != nil {
+		return err
```

Comment:
How does the error look like? Should we make it more friendly

Reply:
<img width="677" alt="image" src="https://github.com/seal-io/walrus/assets/85595852/940cbe6a-d21e-4c45-8450-067dea3b0ef8">
If regexp is not vaild, this is how it looks like , shall we warp this error?

Reply:
fmt.Errorf("invalid filter pattern: %w", err)

Reply:
Then the error will be:
```json
{
  "message": "invalid filter pattern: error parsing regexp: missing closing ]: `[email\\]@example.com`",
  "status": 400,
  "statusText": "Bad Request"
}
```